web3.js/test/transaction.test.ts

```diff
@@ -455,15 +571,8 @@ describe('Transaction', () => {
     );
     transferTransaction.sign(account1);
 
-    let expectedData = Buffer.alloc(4);
-    expectedData.writeInt32LE(4, 0);
-
-    expect(transferTransaction.instructions).to.have.length(2);
-    expect(transferTransaction.instructions[0].programId).to.eql(
-      SystemProgram.programId,
-    );
-    expect(transferTransaction.instructions[0].data).to.eql(expectedData);
-    expect(transferTransaction.recentBlockhash).to.eq(nonce);
+    expect(transferTransaction.instructions).to.have.length(1);
+    expect(transferTransaction.recentBlockhash).to.be.undefined;
```

Comment:
The changes in this test represent the prior behavior that I've changed. No longer does simply calling `sign()` (which in turn calls `compileMessage()`) mutate the `Transaction` instance.